target/linux/ath79/dts/qca9531_tplink_tl-wr902ac-v1.dts

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "tplink,tl-wr902ac-v1", "qca,qca9531";
+	model = "TP-Link TL-WR902AC v1";
+
+	aliases {
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		label-mac-device = &wmac;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan2g {
+			label = "tp-link:green:wlan2g";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		qss {
+			label = "tp-link:green:qss";
+			gpios = <&gpio 10 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "tp-link:green:lan";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&eth0>;
+			linux,default-trigger = "netdev";
+		};
+
+		usb {
+			label = "tp-link:green:usb";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port0>;
+			linux,default-trigger = "usbport";
+		};
+
+		internet {
+			label = "tp-link:green:internet";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		led_system: system {
+			label = "tp-link:green:system";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "WPS button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		};
+
+		sw1 {
+			label = "Mode switch 1";
+			linux,code = <BTN_0>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+
+		sw1 {
+			label = "Mode switch 2";
+			linux,code = <BTN_1>;
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+	
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio_usb_power {
+			gpio-export,name = "tp-link:power:usb";
+			gpio-export,output = <1>;
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@20000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x020000 0x730000>;
+			};
+
+			romfile: partition@750000 {
+				label = "romfile";
+				reg = <0x750000 0x010000>;
```

Comment:
This might be size 0xa0000:
https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/image/generic-tp-link.mk#L527

Despite, in ar71xx it was labelled tplink. Please have a look at similar cases of ar71xx->ath79 ports for the choice of partition label.

Reply:
Indeed it is 0xA0000.